### PR TITLE
Rek react file upload

### PR DESCRIPTION
**Pull request by kilbergr**
## Description

Adding an uploader component for documents. Includes tests for the new component. It currently exists on the dropdown because there's no natural home, but we'll be incorporating it throughout the app, which is why it's in the `shared` dir.

## Reviewer Notes

Can you check out the way confirmation works and whether we should have a different type of alert?

## Verification Steps

* [x] All tests pass.
* [ ] There are no [aXe](https://www.deque.com/products/aXe/) warnings for UI.
* [x] This works in IE.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/155917615) for this change.

## Screenshots
uploadgif

### CODE REVIEW

File: src/appReducer.js [hunk 1]
```diff
@@ -19,6 +20,7 @@ export const appReducer = combineReducers({
   feedback: feedbackReducer,
   signedCertification: signedCertificationReducer,
   DD1299: dd1299Reducer,
+  document: documentReducer,
```
Comment (tinyels) [inline src/appReducer.js]: `document` is a pretty generic name. I'd suggest calling it `uploader`

File: src/shared/AppWrapper/index.jsx [hunk 2]
```diff
@@ -41,6 +42,7 @@ const AppWrapper = () => (
           <PrivateRoute path="/DD1299" component={DD1299} />
           <PrivateRoute path="/moves/:moveId/legalese" component={Legalese} />
           <Route path="/feedback" component={Feedback} />
+          <Route path="/upload" component={Uploader} />
```
Comment (tinyels) [inline src/shared/AppWrapper/index.jsx]: Do we want people to be able to upload files without logging in?

File: src/shared/Uploader/index.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    return (
+      <div className="usa-grid">
+        <input
+          type="file"
+          ref={input => {
+            this.fileInput = input;
+          }}
+        />
+        <button onClick={this.uploadFile}>Upload Now</button>
+        <UploadConfirmation confirmationText={confirmationText} />
```
Comment (tinyels) [inline src/shared/Uploader/index.jsx]: Let's use an `Alert` for this instead to be consistent with USWDS standards. (I have a similar change to the submit a bug feature waiting on review at the moment)

File: src/shared/AppWrapper/index.jsx (hunk 2 above)
Comment (kilbergr) [reply src/shared/AppWrapper/index.jsx]: No. But we're not yet sure where this uploader will be placed. Most likely in forms of various kinds. I'm making it private for now but may make it public again just for the demo. Good catch!

**Pull request merged**